### PR TITLE
Core Forms implementation

### PR DESCRIPTION
**Pull request by mhdostal**
This switches the FormView to use the new Core implementation of Forms and removed FormsPlugin (RIP).

This should not be merged until after we branch to release and the Core Client YAML update PR gets merged.

### CODE REVIEW

**Comment by mhdostal** [comment]
This is ready for review. I left the `FormViewModel` for now; it will need to be updated/cleaned up now that the `FeatureForm` is handling some "model" duties.

File: Examples/Examples/FormExampleView.swift [hunk 1]
```diff
@@ -29,17 +29,20 @@ struct FormExampleView: View {
     
     /// The form view model provides a channel of communication between the form view and its host.
     @StateObject private var formViewModel = FormViewModel()
-    
+    @State private var featureForm: FeatureForm?
+
```
Comment (dfeinzimer) [inline Examples/Examples/FormExampleView.swift:33]:
```suggestion
    
    /// The form being edited in the form view.
    @State private var featureForm: FeatureForm?
    
```

Or some similar doc comment.

File: Examples/Examples/FormExampleView.swift [hunk 2]
```diff
@@ -29,17 +29,20 @@ struct FormExampleView: View {
...
                 }
                 .task(id: identifyScreenPoint) {
-                    if let feature = await identifyFeature(with: mapViewProxy) {
-                        formViewModel.startEditing(feature)
-                        isPresented = true
+                    if let feature = await identifyFeature(with: mapViewProxy),
+                       let formDefinition = (feature.table?.layer as? FeatureLayer)?.featureFormDefinition {
+                        featureForm = FeatureForm(feature: feature, definition: formDefinition)
+//                        formViewModel.startEditing(feature)
+                        isPresented = featureForm != nil // true
```
Comment (dfeinzimer) [inline Examples/Examples/FormExampleView.swift:45]:
```suggestion
                        isPresented = featureForm != nil
```

File: Sources/ArcGISToolkit/Components/Form/FormInputs/DateTimeEntry.swift [hunk 3]
```diff
@@ -11,15 +11,16 @@
...
 import SwiftUI
+import ArcGIS
 
 struct DateTimeEntry: View {
     @Environment(\.formElementPadding) var elementPadding
     
     /// The model for the ancestral form view.
     @EnvironmentObject var model: FormViewModel
-    
+    private var featureForm: FeatureForm?
```
Comment (dfeinzimer) [inline Sources/ArcGISToolkit/Components/Form/FormInputs/DateTimeEntry.swift:22]:
```suggestion
    
    private var featureForm: FeatureForm?
    
```

File: Sources/ArcGISToolkit/Components/Form/FormInputs/DateTimeEntry.swift [hunk 4]
```diff
@@ -30,16 +31,17 @@ struct DateTimeEntry: View {
...
-    private let element: FieldFeatureFormElement
+    private let element: FieldFormElement
     
     /// The input configuration of the field.
-    private let input: DateTimePickerFeatureFormInput
+    private let input: DateTimePickerFormInput
     
     /// Creates a view for a date and time (if applicable) entry.
     /// - Parameters:
     ///   - element: The field's parent element.
```
Comment (dfeinzimer) [inline Sources/ArcGISToolkit/Components/Form/FormInputs/DateTimeEntry.swift:41]:
```suggestion
    ///   - featureForm: <#featureForm description#>
    ///   - element: The field's parent element.
```

File: Sources/ArcGISToolkit/Components/Form/FormInputs/MultiLineTextEntry.swift [hunk 5]
```diff
@@ -11,16 +11,17 @@
...
 
 /// A view for text entry spanning multiple lines.
 struct MultiLineTextEntry: View {
     @Environment(\.formElementPadding) var elementPadding
     
     /// The model for the ancestral form view.
     @EnvironmentObject var model: FormViewModel
-    
+    private var featureForm: FeatureForm?
+
```
Comment (dfeinzimer) [inline Sources/ArcGISToolkit/Components/Form/FormInputs/MultiLineTextEntry.swift:24]:
```suggestion
    
    /// <#Description#>
    private var featureForm: FeatureForm?
    
```

File: Sources/ArcGISToolkit/Components/Form/FormInputs/MultiLineTextEntry.swift [hunk 6]
```diff
@@ -35,16 +36,17 @@ struct MultiLineTextEntry: View {
...
-    private let element: FieldFeatureFormElement
+    private let element: FieldFormElement
     
     /// The input configuration of the field.
-    private let input: TextAreaFeatureFormInput
+    private let input: TextAreaFormInput
     
     /// Creates a view for text entry spanning multiple lines.
     /// - Parameters:
     ///   - element: The field's parent element.
```
Comment (dfeinzimer) [inline Sources/ArcGISToolkit/Components/Form/FormInputs/MultiLineTextEntry.swift:46]:
```suggestion
    ///   - featureForm: <#featureForm description#>
    ///   - element: The field's parent element.
```

File: Sources/ArcGISToolkit/Components/Form/FormInputs/SingleLineTextEntry.swift [hunk 7]
```diff
@@ -11,13 +11,14 @@
...
-import FormsPlugin
 import SwiftUI
+import ArcGIS
 
 /// A view for single line text entry.
 struct SingleLineTextEntry: View {
     @Environment(\.formElementPadding) var elementPadding
-    
+    private var featureForm: FeatureForm?
+
```
Comment (dfeinzimer) [inline Sources/ArcGISToolkit/Components/Form/FormInputs/SingleLineTextEntry.swift:21]:
```suggestion
    
    /// <#Description#>
    private var featureForm: FeatureForm?
    
```

File: Sources/ArcGISToolkit/Components/Form/FormInputs/SingleLineTextEntry.swift [hunk 8]
```diff
@@ -28,16 +29,17 @@ struct SingleLineTextEntry: View {
...
-    private let element: FieldFeatureFormElement
+    private let element: FieldFormElement
     
     /// The input configuration of the field.
-    private let input: TextBoxFeatureFormInput
+    private let input: TextBoxFormInput
     
     /// Creates a view for single line text entry.
     /// - Parameters:
     ///   - element: The field's parent element.
```
Comment (dfeinzimer) [inline Sources/ArcGISToolkit/Components/Form/FormInputs/SingleLineTextEntry.swift:39]:
```suggestion
    ///   - featureForm: <#featureForm description#>
    ///   - element: The field's parent element.
```

File: Sources/ArcGISToolkit/Components/Form/FormView.swift [hunk 9]
```diff
@@ -22,49 +21,39 @@ public struct FormView: View {
     
     /// The model for this form view.
     @EnvironmentObject var model: FormViewModel
+    private var featureForm: FeatureForm?
```
Comment (dfeinzimer) [inline Sources/ArcGISToolkit/Components/Form/FormView.swift:24]:
```suggestion
    
    /// <#Description#>
    private var featureForm: FeatureForm?
```

File: Sources/ArcGISToolkit/Components/Form/FormView.swift [hunk 10]
```diff
@@ -22,49 +21,39 @@ public struct FormView: View {
...
     
     /// Initializes a form view.
-    public init() {}
+    public init(featureForm: FeatureForm?) {
```
Comment (dfeinzimer) [inline Sources/ArcGISToolkit/Components/Form/FormView.swift:27]:
```suggestion
    /// - Parameter featureForm: <#featureForm description#>
    public init(featureForm: FeatureForm?) {
```

**Review by dfeinzimer — COMMENTED** [review]
Mainly just small formatting things. Feel free to leave unchanged. Exciting to see the changeover going in.

**Review by dfeinzimer — APPROVED** [review]
Looks good!

**Pull request merged**